# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/bindings/ToJSValue.h
# Commit: d88b294e0a5e
# Full Hash: d88b294e0a5e00a0ca739037ffa1452e39879382
# Author: Zibi Braniecki <zbraniecki@mozilla.com>
# Date: 2021-08-03 22:01:19
# Regressor Bug: 1613705
# File Overlap Count: 1
# Description:
#   Bug 1613705 - [localization] part2: Switch Localization class to use localization-ffi. r=emilio,nika
#   
#   Depends on D104788
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D104789
# ==============================================================================

diff -r e6bebff87544 -r d88b294e0a5e dom/bindings/ToJSValue.h
--- a/dom/bindings/ToJSValue.h	Tue Aug 03 05:52:00 2021 +0000
+++ b/dom/bindings/ToJSValue.h	Tue Aug 03 05:52:00 2021 +0000
@@ -435,6 +435,17 @@
   return true;
 }
 
+template <typename T>
+[[nodiscard]] bool ToJSValue(JSContext* aCx, const Nullable<T>& aArgument,
+                             JS::MutableHandle<JS::Value> aValue) {
+  if (aArgument.IsNull()) {
+    aValue.setNull();
+    return true;
+  }
+
+  return ToJSValue(aCx, aArgument.Value(), aValue);
+}
+
 }  // namespace dom
 }  // namespace mozilla
 